packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -507,6 +528,21 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
     return Future<void>.value();
   }
 
+  final Map<PreHotRestartCallback, String?> _hotRestartCallbacks = <PreHotRestartCallback, String?>{};
+
+  /// Register a callback that will be invoked before a hot restart is called.
+  ///
+  /// In non-debug modes this method is a no-op.
+  ///
+  /// This can be used to release native resources acquired through platform
+  /// channels or `dart:ffi`.
+  void registerHotRestartCallback(PreHotRestartCallback callback, {String? debugLabel}) {
```

Comment:
you can simplify some of the code here by making `debugLabel` have a default value ("unknown" or something)

Reply:
done